skribenten-web/frontend/src/routes/saksnummer_.$saksId.brevvelger.$templateId.tsx

```diff
@@ -428,14 +437,48 @@ function SelectLanguage({ letterTemplate }: { letterTemplate: LetterMetadata })
   );
 }
 
+function SelectEnhet() {
+  const enheterQuery = useQuery(getEnheter);
+  const { enhetsId } = Route.useSearch();
+  const { register, setValue, formState } = useFormContext();
+  const navigate = useNavigate({ from: Route.fullPath });
+  const options = enheterQuery.data ?? [];
+  useEffect(() => {
+    if (enhetsId) {
+      setValue("enhetsId", enhetsId);
+    }
+  }, [enhetsId, setValue]);
+  return (
+    <Select
+      {...register("enhetsId")}
+      error={formState.errors.enhetsId?.message?.toString()}
+      label="Avsenderenhet"
+      onChangeCapture={(element) => {
+        navigate({
+          search: (s) => ({ ...s, enhetsId: element.currentTarget.value }),
+          replace: true,
+        });
+      }}
+      size="medium"
+      value={enhetsId}
+    >
+      <option value={""}>Velg enhet</option>
```

Comment:
Vil validering her sørge for at "Velg enhet" ikke er et gyldig valg?

Reply:
Ja. Zod sørger for det siden value for "Velg enhet" er "" og valideringen er " enhetsId: z.string().min(1, "Obligatorisk"),".